tests/server/test_compilerservice.py

```diff
@@ -843,8 +841,14 @@ async def old_and_new_compile_report(server_with_frequent_cleanups, environment_
         "name": "Recompiling configuration model",
         "compile": compile_id_new,
     }
-    await Report(**report1).insert()
-    await Report(**report2).insert()
+
+    async with Compile.get_connection() as con:
+        async with con.transaction():
+            await Compile(**old_compile).insert()
+            await Compile(**new_compile).insert()
+            await Report(**report1).insert()
+            await Report(**report2).insert()
```

Comment:
You need to pass on the connection object to the insert() method. Otherwise the insert() method will create a new connection, causing it to execute outside of the transaction.
```suggestion
            await Compile(**old_compile).insert(connection=con)
            await Compile(**new_compile).insert(connection=con)
            await Report(**report1).insert(connection=con)
            await Report(**report2).insert(connection=con)
```